friendly/src/main/java/com/google/sps/servlets/RecommendationServlet.java

```diff
@@ -18,21 +27,28 @@
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-        String groupId = request.getParameter("groupId");
-        String restaurantName = request.getParameter("restaurantName");
-        String location = request.getParameter("location");
-        String userEmail = request.getParameter("userEmail");
+        UserService userService = UserServiceFactory.getUserService();
+
+        // user not logged in, can not post recommendations
+        if (!userService.isUserLoggedIn()){
```

Comment:
Actually do we need this? I thought if the user is not logged in, the user will not even be able to see this form?

Reply:
just precautionary measures in case some other service tries to post into ours HAHAHA

Reply:
Defensive measure is good. A bad user can manually compose http request or some other page may fail to do permission check.